tasks/testChainAdapter.ts

```diff
@@ -0,0 +1,68 @@
+import assert from "assert";
+import { getMnemonic } from "@uma/common";
+import { task } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { CHAIN_IDs, TOKEN_SYMBOLS_MAP } from "../utils/constants";
+
+// Chain adapter names are not 1:1 consistent with chain names, so some overrides are needed.
+const chains = {
+  [CHAIN_IDs.ARBITRUM]: "Arbitrum_Adapter",
+  [CHAIN_IDs.WORLD_CHAIN]: "WorldChain_Adapter",
+  [CHAIN_IDs.ZK_SYNC]: "ZkSync_Adapter",
+};
+
+task("testChainAdapter", "Verify a chain adapter")
+  .addParam("chain", "chain ID of the adapter being tested")
+  .addParam("token", "Token to bridge to the destination chain")
+  .setAction(async function (args, hre: HardhatRuntimeEnvironment) {
+    const { deployments, ethers, getChainId, network } = hre;
+    const provider = new ethers.providers.StaticJsonRpcProvider(network.config.url);
+    const signer = new ethers.Wallet.fromMnemonic(getMnemonic()).connect(provider);
+
+    const hubChainId = await getChainId();
+    const { address: hubPoolAddress, abi: hubPoolAbi } = await deployments.get("HubPool");
+    const hubPool = new ethers.Contract(hubPoolAddress, hubPoolAbi, provider);
+    const spokeChainId = parseInt(args.chain);
+
+    const [spokeName] = Object.entries(CHAIN_IDs).find(([, chainId]) => chainId === spokeChainId) ?? [];
+    assert(spokeName, `Could not find any chain entry for chainId ${spokeChainId}.`);
+    const adapterName =
+      chains[spokeChainId] ?? `${spokeName[0].toUpperCase()}${spokeName.slice(1).toLowerCase()}_Adapter`;
+
+    const { address: adapterAddress, abi: adapterAbi } = await deployments.get(adapterName);
+    const adapter = new ethers.Contract(adapterAddress, adapterAbi, provider);
+    // const spokeAddress = getDeployedAddress("SpokePool", spokeChainId, true);
+    const tokenSymbol = args.token.toUpperCase();
+    const tokenAddress = TOKEN_SYMBOLS_MAP[tokenSymbol].addresses[hubChainId];
+
+    // For USDC this will resolve to native USDC on CCTP-enabled chains.
+    const l2Token = await hubPool.poolRebalanceRoute(spokeChainId, tokenAddress);
+    console.log(`Resolved ${tokenSymbol} l2 token address on chain ${spokeChainId}: ${l2Token}.`);
+
+    const erc20 = (await ethers.getContractFactory("ExpandedERC20")).attach(tokenAddress);
+    const amount = 1_000_000;
```

Comment:
Amount should be configurable on the cmdline, passed in as the human-readable form (i.e. 1 for 1 USDC), and then promoted to the necessary number of decimals based on the token being bridged.